File: BackEnd/src/main/java/org/example/dto/VehiculoDTO.java
package org.example.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class VehiculoDTO {
    private Long idVehiculo;
    private String matricula;
    private String marca;
    private String modelo;
    private Integer año;
    private String color;
    private Integer kilometraje;
    private String observaciones;
    private String medidasTomadas;
    private String estadoRevision;

    // Imagen en Base64 para enviar/recibir
    private String imagenBase64;
    
    private Long idCliente;
    private String nombreCliente;
    private String emailCliente;
    private Long idMecanico;
    private String nombreMecanico;
    private String emailMecanico;
    private LocalDateTime fechaIngreso;
}

File: Android/app/src/main/java/com/example/autolog/core/data/remote/dto/vehiculo/VehiculoDTO.kt
package com.example.autolog.core.data.remote.dto.vehiculo

import com.google.gson.annotations.SerializedName

data class VehiculoDTO(
    @SerializedName("idVehiculo")
    val idVehiculo: Long,
    
    @SerializedName("matricula")
    val matricula: String,
    
    @SerializedName("marca")
    val marca: String,
    
    @SerializedName("modelo")
    val modelo: String,
    
    @SerializedName("anio")
    val anio: Int,
    
    @SerializedName("color")
    val color: String?,
    
    @SerializedName("kilometraje")
    val kilometraje: Int?,
    
    @SerializedName("observaciones")
    val observaciones: String?,
    
    @SerializedName("medidasTomadas")
    val medidasTomadas: String?,
    
    @SerializedName("estadoRevision")
    val estadoRevision: String?,
    
    @SerializedName("imagenBase64")
    val imagenBase64: String?,
    
    @SerializedName("idCliente")
    val idCliente: Long,
    
    @SerializedName("nombreCliente")
    val nombreCliente: String?,
    
    @SerializedName("emailCliente")
    val emailCliente: String?,
    
    @SerializedName("idMecanico")
    val idMecanico: Long?,
    
    @SerializedName("nombreMecanico")
    val nombreMecanico: String?,
    
    @SerializedName("emailMecanico")
    val emailMecanico: String?,
    
    @SerializedName("fechaIngreso")
    val fechaIngreso: String?
)
